pkgs/applications/networking/cluster/opentofu/default.nix

```diff
@@ -0,0 +1,173 @@
+{ stdenv
+, lib
+, buildGoModule
+, fetchFromGitHub
+, makeWrapper
+, coreutils
+, runCommand
+, runtimeShell
+, writeText
+, terraform-providers
+, installShellFiles
+}:
+
+let
+  package =  buildGoModule rec {
+    pname = "opentofu";
+    version = "unstable-2023-09-20";
+
+    src = fetchFromGitHub {
+      owner = "opentofu";
+      repo = "opentofu";
+      rev = "922f61d84742804dc4357a90ae56c20a07a80683";
+      hash = "sha256-UPv+xvKZOaIR+MWtZfm5XA7KPCL8x4iD0fAqQFkkZFE=";
+    };
+    vendorHash = "sha256-7GLmgdESm28PTB+aK/9vjCk0JGA2dA5KUCXr2a4iLEw=";
+    ldflags = [ "-s" "-w" ];
+
+    postConfigure = ''
+        # speakeasy hardcodes /bin/stty https://github.com/bgentry/speakeasy/issues/22
+        substituteInPlace vendor/github.com/bgentry/speakeasy/speakeasy_unix.go \
+          --replace "/bin/stty" "${coreutils}/bin/stty"
+      '';
+
+    nativeBuildInputs = [ installShellFiles ];
+    patches = [ ./provider-path-0_15.patch ];
+
+    passthru = {
+      inherit plugins withPlugins;
+      tests = { inherit opentofu_plugins_test; };
+    };
+
+    postInstall = ''
+        # https://github.com/posener/complete/blob/9a4745ac49b29530e07dc2581745a218b646b7a3/cmd/install/bash.go#L8
+        installShellCompletion --bash --name opentofu <(echo complete -C opentofu opentofu)
+      '';
+
+    preCheck = ''
+        export HOME=$TMPDIR
+        export TF_SKIP_REMOTE_TESTS=1
+      '';
+
+    subPackages = [ "." ];
+
+    meta = with lib; {
+      description =
+        "Tool for building, changing, and versioning infrastructure";
+      homepage = "https://opentofu.org/";
+      changelog = "https://github.com/opentofu/opentofu/blob/${version}/CHANGELOG.md";
```

Comment:
Currently, I'm pretty sure this evaluates to https://github.com/opentofu/opentofu/blob/unstable-2023-09-20/CHANGELOG.md

It probably should be the `src.revision` instead https://github.com/opentofu/opentofu/blob/922f61d84742804dc4357a90ae56c20a07a80683/CHANGELOG.md